src/terraform/resources/terraform_qsfs_on_microvm.md

```diff
@@ -1,12 +1,84 @@
-<h1> QSFS on Micro VM </h1>
+<h1> QSFS on Micro VM with Terraform</h1>
 
 ![ ](../advanced/img/terraform_.png)
 
-## QSFS Example on Micro VM
+<h2>Table of Contents</h2>
 
-We present below an example of a QSFS deployment on a microVM. Note that you will need to replace the line `SSH_KEY = "PUT YOUR SSH KEY HERE"` with your own SSH key.
+- [Introduction](#introduction)
+- [Prerequisites](#prerequisites)
+- [Find a 3Node](#find-a-3node)
+- [Create the Terraform Files](#create-the-terraform-files)
+- [Deploy the Micro VM with Terraform](#deploy-the-micro-vm-with-terraform)
+- [SSH into the 3Node](#ssh-into-the-3node)
+- [Questions and Feedback](#questions-and-feedback)
+
+***
+
+## Introduction
+
+In this ThreeFold Guide, we will learn how to deploy a Quantum Safe File System (QSFS) deployment with Terraform. The main template for this example can be found [here](https://github.com/threefoldtech/terraform-provider-grid/blob/development/examples/resources/qsfs/main.tf). 
+
+***
+## Prerequisites
+
+In this guide, we will be using Terraform to deploy a QSFS workload on a micro VM that runs on the TFGrid. Make sure to have the latest Terraform version.
+
+* [Install Terraform](https://developer.hashicorp.com/terraform/downloads)
+
+***
+
+
+## Find a 3Node
+
+We want to find a proper 3Node to deploy our workload. For this guide, we want a 3Node with at least 15GB of storage, 1 vcore and 512MB of RAM, which are the minimum specifications for a micro VM on the TFGrid. We are also looking for a 3Node with a public IPv4 address.
+
+We show here how to find a suitable 3Node using the ThreeFold Explorer.
+
+* Go to the ThreeFold Grid's [Explorer](https://dashboard.grid.tf/explorer/nodes) (Main Net)
+* Find a 3Node with suitable resources for the deployment and take note of its node ID on the leftmost column `ID`
+* For proper understanding, we give further information on some relevant columns:
+  * `ID` refers to the node ID
+  * `Free Public IPs` refers to available IPv4 public IP addresses
+  * `HRU` refers to HDD storage
+  * `SRU` refers to SSD storage
+  * `MRU` refers to RAM (memory)
+  * `CRU` refers to virtual cores (vcores)
+* To quicken the process of finding a proper 3Node, you can narrow down the search by adding filters:
+  * At the top left of the screen, in the `Filters` box, select the parameter(s) you want.
+  * For each parameter, a new field will appear where you can enter a minimum number requirement for the 3Nodes.
+    * `Free SRU (GB)`: 15
+    * `Free MRU (GB)`: 1
+    * `Total CRU (Cores)`: 1
+    * `Free Public IP`: 2
+      * Note: if you want a public IPv4 address, it is recommended to set the parameter `FREE PUBLIC IP` to at least 2 to avoid false positives. This ensures that the shown 3Nodes have viable IP addresses.
+
+Once you've found a proper node, take node of its node ID. You will need to use this ID when creating the Terraform files.
+
+***
+
+## Create the Terraform Files
+
```

Comment:
Can we also mention that all of these steps are already created in the Terraform repo and can be easily run by cloning it and simply running this [command](https://github.com/threefoldtech/terraform-provider-grid/tree/development#using-the-provider) ?

Reply:
Excellent idea @A-Harby !
I pushed a new commit with what you asked here. I also did a quick test to make sure it all works fine.

Thanks.